map/chart_generator.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "indexer/feature_altitude.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "std/deque.hpp"
+#include "std/vector.hpp"
+
+/// \brief fills uniformAltitudeDataM with altitude data which evenly distributed by
+/// |resultPointCount| points. |distanceDataM| and |altitudeDataM| form a curve of route altitude.
+/// This method is used to generalize and evenly distribute points of the chart.
+void NormalizeChartData(deque<double> const & distanceDataM,
```

Comment:
From my point of view it's not a problem when a getter just appends to the provided vector. This property sometimes is very useful, for example when you need to collect results of several calls in the same vector. On the other hand, this approach makes it difficult to throw away current contents of the vector and fill it with results of the latest call. In any case, the behavior should be explained explicitly in the comment to the function.

It's up to you what to do here, but if you want to throw away current contents of the provided vectors, why not to return them by value? In other words, to prevent misunderstanding, I propose you to do following:

Instead of
```c++
void Foo(vector & v);
```

do this:
```c++
vector Foo();
```

No place for misunderstanding, right?